src/main/java/org/toradocu/extractor/Tag.java

```diff
@@ -1,56 +1,159 @@
+        case "THROWS":
+          return "@throws";
         default:
-          throw new IllegalStateException("The value " + name() + " has no string representation.");
+          throw new IllegalStateException("The value " + label + " has no string representation.");
       }
     }
   }
 
+  /** The comment of this tag. */
+  private Comment comment;
+
+  /** The kind of this tag (e.g., @throws, @param). */
+  private final Kind kind;
+
+  /**
+   * Specification generated from the comment of this {@code Tag}. {@code null} if Toradocu failed
+   * to generate a specification or if comment translation not yet attempted.
+   */
+  private S specification;
+
+  /**
+   * Constructs a {@code Tag} of the specific kind, with the given comment.
+   *
+   * @param kind the comment kind. See {@code Tag.Kind} for the available kinds.
+   * @param comment the comment associated with the exception
+   * @throws NullPointerException if either kind or comment is null
+   */
+  Tag(Kind kind, Comment comment) {
+    Checks.nonNullParameter(kind, "kind");
+    Checks.nonNullParameter(comment, "comment");
+    this.kind = kind;
+    this.comment = comment;
+  }
+
   /**
    * Returns the kind of this tag (e.g., @throws, @param).
    *
    * @return the kind of this tag
    */
-  Kind getKind();
+  public Kind getKind() {
+    return kind;
+  }
 
   /**
-   * Returns the translated Java boolean condition for this tag as an optional which is empty if
-   * translation has not been attempted yet.
+   * Returns the specification that represents the translation for this tag. {@code null} if the
+   * translation has not been attempted yet or no translation has been generated.
    *
-   * @return the translated conditions for this tag if translation attempted, else empty optional
+   * @return the translation for this tag. {@code null} if the translation has not been attempted
+   *     yet or no translation has been generated.
    */
-  Optional<String> getCondition();
+  public S getSpecification() {
+    return specification;
+  }
 
   /**
-   * Sets the translated condition for this tag to the given condition.
+   * Sets the specification (translation) for this tag to the given specification.
    *
-   * @param condition the translated condition for this tag (as a Java boolean condition)
-   * @throws NullPointerException if condition is null
+   * @param specification the comment translation for this tag (a specification)
    */
-  void setCondition(String condition);
+  public void setSpecification(S specification) {
+    this.specification = specification;
+  }
 
   /**
    * Returns the comment associated with the exception in this tag.
    *
    * @return the comment associated with the exception in this tag
    */
-  String getComment();
+  public Comment getComment() {
+    return comment;
+  }
+
+  /**
+   * Sets the comment for this tag.
+   *
+   * @param comment the comment for this tag
+   * @throws NullPointerException if comment is null
```

Comment:
This comment is incorrect:  no exception is thrown.
However, per my previous comments, it's better to document the restrictions on the parameters and not over-specify by requiring a particular exception to be thrown.